src/compiler/scala/tools/nsc/PhaseAssembly.scala

```diff
@@ -237,18 +242,18 @@ object DependencyGraph {
       for (p <- phases) {
         require(p.phaseName.nonEmpty, "Phase name must be non-empty.")
         def checkConstraint(name: String, constraint: String): Boolean =
-          phases.exists(_.phaseName == name).tap(ok => if (!ok) graph.warning(s"No phase `$name` for ${p.phaseName}.$constraint"))
+          graph.components.contains(name).tap(ok => if (!ok) graph.warning(s"No phase `$name` for ${p.phaseName}.$constraint"))
```

Comment:
This quadratic lookup is where `PhaseAssemblyTest.multipleRunsRightAfter` was spending all of its time